File: appmonet-sdk/appmonet-sdk-dfp/src/main/java/com/monet/bidder/DFPAdServerWrapper.java
package com.monet.bidder;

import androidx.annotation.NonNull;
import com.monet.bidder.auction.AuctionRequest;

/**
 * Created by nbjacob on 6/26/17.
 */

class DFPAdServerWrapper implements AdServerWrapper {
  private SdkManager sdkManager;

  public void setSdkManager(SdkManager sdkManager) {
    this.sdkManager = sdkManager;
  }

  @NonNull
  @Override
  public AdServerAdRequest newAdRequest(@NonNull AuctionRequest auctionRequest) {
    return AdRequestFactory.fromAuctionRequest(sdkManager.isPublisherAdView, auctionRequest);
  }

  @NonNull
  @Override
  public AdServerAdRequest newAdRequest(@NonNull AuctionRequest auctionRequest,
      @NonNull Type type) {
    return newAdRequest(auctionRequest);
  }

  @NonNull
  @Override
  public AdServerAdRequest newAdRequest() {
    return AdRequestFactory.createEmptyRequest(sdkManager.isPublisherAdView);
  }

  @NonNull
  @Override
  public AdSize newAdSize(Integer width, Integer height) {
    return new AdSize(width, height);
  }
}

File: appmonet-sdk/appmonet-sdk-dfp/src/main/java/com/monet/bidder/DFPAdServerWrapper.kt
package com.monet.bidder

import android.content.Context
import com.monet.AdServerAdRequest
import com.monet.AdServerWrapper
import com.monet.AdType
import com.monet.adview.AdSize
import com.monet.auction.AuctionRequest
import com.monet.bidder.AdRequestFactory.createEmptyRequest
import com.monet.bidder.AdRequestFactory.fromAuctionRequest

/**
 * Created by nbjacob on 6/26/17.
 */
internal class DFPAdServerWrapper(private val context: Context) : AdServerWrapper {
  private var sdkManager: SdkManager? = null
  fun setSdkManager(sdkManager: SdkManager?) {
    this.sdkManager = sdkManager
  }

  override fun newAdRequest(auctionRequest: AuctionRequest): AdServerAdRequest {
    return fromAuctionRequest(sdkManager!!.isPublisherAdView, auctionRequest)
  }

  override fun newAdRequest(
    auctionRequest: AuctionRequest,
    type: AdType
  ): AdServerAdRequest {
    return newAdRequest(auctionRequest)
  }

  override fun newAdRequest(): AdServerAdRequest {
    return createEmptyRequest(sdkManager!!.isPublisherAdView)
  }

  override fun newAdSize(
    width: Int,
    height: Int
  ): AdSize {
    return AdSize(context, width, height)
  }
}